Minor fixes

stefancarlton opened the pull request:
*Mysql Instance name*

In `mysql_master.rb` and `mysql_slave.rb` the attribute service name was ignored and 'chef' was hardcoded, there was no variables available (e.g. version, port, bind address etc). It made sense therefore to keeps things standard and bugfix at the same time - thus call default.rb

*Optional Mysql install*

To support custom MySQL installs (e.g. with more attributes like data_dir) the automatic inclusion of mysql-multi::default.rb was removed.

martinb3 commented:
This looks okay to me, but I wonder if we should emit some sort of warning to warn people that it's not backwards compatible or that the recipes have changed.

stefancarlton commented:
Good point regarding backwards compatibility - removing the installation also caused issues with calling restart as at compile time the service wasn't known. To fix this I added an attribute / recipe call which means users (aka me) can override and define their own installation should they desire otherwise remain with the backwards compatibility.

*Other fixes*
I've also fixed a few variable differences in "master_ip" vs "master", "slave_ip" vs "slaves" and migrated the root password saving to the `default.rb`

cybermerc commented:
Overall I really like these changes, we had planned to phase out the recipes for a libraries only approach so we did not spend the time on these that they needed. The fact that they are breaking changes does not bother me too much since we just did a major update which was not backwards compatible anyways (to version 2.0). thanks for your submission we will get them incorporated as soon as possible.
